tidoop-mr-lib-api/src/server.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2015 Telefonica Investigación y Desarrollo, S.A.U
+ *
+ * This file is part of fiware-tidoop (FI-WARE project).
+ *
+ * fiware-tidoop is free software: you can redistribute it and/or modify it under the terms of the GNU Affero
+ * General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ * fiware-tidoop is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+ * implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with fiware-tidoop. If not, see
+ * http://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License please contact with
+ * francisco dot romerobueno at telefonica dot com
+ */
+
+/**
+ * Http server for "Tidoop MR job library" REST API
+ *
+ * Author: frb
+ */
+
+var Hapi = require('hapi');
+
+// create a server with a host and port
+var server = new Hapi.Server();
+var config = require('../conf/tidoop-mr-lib-api.json');
+
+server.connection({ 
+    host: 'localhost',
+    port: config.port
+});
+
+// add the route
+server.route({
+    method: 'GET',
+    path:'/version',
+    handler: function (request, reply) {
+        console.log("Request: GET /version")
+        var pjson = require('../package.json');
+        var response = '{version: ' + pjson.version + '}';
+        console.log("Response: " + response);
+        reply(response);
+    }
+});
+
+// start the server
+server.start();
+console.log("tidoop-mr-lib-api running at http://localhost:" + config.port);
```

Comment:
As it is common practice in Node, most of the function calls are asynchronous and notify their ending via a callback function. `server.start()` is an example of this as you can see at http://hapijs.com/api#serverstartcallback

Consequently, you may be logging `tidoop-mr-lib-api running at http://localhost:6060` to console before the server really started or even in case the starting of the server had some problem. The recommended way to deal with this situation would be:
````
server.start(function(err) {
  if(err) {
    // Deal with the error
    return console.log('Some error occurred during the starting of the Hapi server: " + err);
    // Using return in the previous line makes the app leave the callback and any remaining code from executing.
  }
  console.log("tidoop-mr-lib-api running at http://localhost:" + config.port);
});
```